### PR TITLE
Update hl._std.sys.thread.Lock

### PR DESCRIPTION
**Pull request by ConstNW**
haxe implementation of the lock bindings(HaxeFoundation/hashlink/pull/292) suggested by @ncannasse at #8699

### CODE REVIEW

**Comment by Aurel300** [comment]
Please remove the docstrings from the `hl` implementation.

File: std/hl/_std/sys/thread/Lock.hx [hunk 1]
```diff
@@ -22,29 +22,73 @@
 
 package sys.thread;
 
+#if doc_gen
 @:coreApi
-class Lock {
-	var deque:sys.thread.Deque<Bool>;
+extern class Lock {
+	public function new():Void;
```
Comment (Gama11) [inline std/hl/_std/sys/thread/Lock.hx]: `public` is redundant in externs.

**Comment by ncannasse** [comment]
This cannot be merged as-it as we need to be able to set the minimal HL version to 1.11, the way we did for Date utc_inf

**Comment by ConstNW** [comment]
@ncannasse is it ready to merge?

**Comment by ncannasse** [comment]
I think there's still an issue given for documentation generation since we cannot have a std class that is sometimes an abstract on some platforms. please confirm @RealyUniqueName

**Comment by RealyUniqueName** [comment]
Yeah, that's bad. Especially given that in the doc we will have it listed as a class, but in an actuall code it will be an abstract.

I suggest to change it to something like this:
```haxe
class Lock {
  var native:hl.Abstract<"hl_lock">;
  <...>
}
```

**Comment by ncannasse** [comment]
@RealyUniqueName ok for 4.0 branch

**Comment by RealyUniqueName** [comment]
Done

**Pull request merged**